src/visitable.h

```diff
@@ -58,6 +59,16 @@ class visitable
 
         /** Returns true if any item (including those within a container) matches the filter */
         bool has_item_with( const std::function<bool( const item & )> &filter ) const;
+
+        /**
+         * Removes items contained by this instance which match the filter
+         * @note if this instance itself is an item it will not be considered by the filter
+         * @param filter a UnaryPredicate which should return true if the item is to be removed
+         * @param count maximum number of items to remove or negative for unlimited. A count of zero is a no-op
+         * @return any items removed (items counted by charges are not guaranteed to be stacked)
+         */
+        std::list<item> remove_items_with( const std::function<bool( const item & )> &filter,
+                                           int count = -1 );
```

Comment:
`INT_MAX` or `std::numeric_limits<int>::max()` would be a better default value for the `count` parameter. It would be more obvious about its behavior and it does not require special cases inside the function (checking for `== -1`). You can even add a generic check for `count >= 0`.